packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate.sh

```diff
@@ -0,0 +1,74 @@
+#!/bin/sh
+
+# check if system is registered
+subscription-manager identity > /dev/null
+if [ $? -ne 0 ]; then
+    echo "Please register with subscription-manager first, then re-run `katello-pull-provider-migrate.sh`."
+    exit 1
+fi
+
+echo "Getting configuration from subscription-manager..."
+KPPTEMPFILE=/tmp/kpp_tempfile
+subscription-manager config --list > $KPPTEMPFILE
+CONSUMER_CERT_DIR=$(grep 'consumercertdir' $KPPTEMPFILE | cut -d= -f2 | xargs | sed 's/[][]//g')
+CERT_FILE=$CONSUMER_CERT_DIR/cert.pem
+KEY_FILE=$CONSUMER_CERT_DIR/key.pem
+CA_FILE=$(grep 'repo_ca_cert = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+SERVER_NAME=$(grep 'hostname = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+
+# fail if no server name, cert dir, or ca file
+if [ -z "$SERVER_NAME" ] || [ -z "$CONSUMER_CERT_DIR" ] || [ -z "$CA_FILE" ]; then
+    echo "Unable to determine config from `subscription-manager config --list`; exiting"
+    exit 1
+fi
+
+# fail if client is not registered to a Katello
+if ! grep -q 'prefix = \/rhsm' $KPPTEMPFILE; then
+    echo "Client is not registered to Katello; exiting"
+    exit 2
+fi
+
+rm -f $KPPTEMPFILE
+
+# set SYSCONFDIR to /etc if it is not set
+if [ -z "$SYSCONFDIR" ]; then
+    SYSCONFDIR=/etc
+fi
+
+# see if /etc/yggdrasil/config.toml exists
+CONFIGTOML=$SYSCONFDIR/yggdrasil/config.toml
+if [ -f $CONFIGTOML ]; then
+    # make a backup of CONFIGTOML
+    cp $CONFIGTOML $CONFIGTOML.bak
+    cat <<EOF > $CONFIGTOML
+# yggdrasil global configuration settings written by katello-pull-provider-migrate
+
+broker = ["tcp://$SERVER_NAME:1883"]
+cert-file = "$CERT_FILE"
+key-file = "$KEY_FILE"
+ca-root = ["$CA_FILE"]
+log-level = "error"
+EOF
+
+else
+    echo "$SYSCONFDIR/yggdrasil/config.toml not found! Did `yum install yggdrasil` succeed?"
+    exit 1
+fi
+
+# start the yggdrasild service
+echo "Starting yggdrasild..."
+systemctl start yggdrasild
```

Comment:
@adamruzicka thinking about your email reply, should this be `systemctl enable yggdrasild`?

Reply:
More like `systemctl enable --now yggdrasild`, although I'm not sure if that's available on all supported platforms

```
systemctl enable yggdrasild
systemctl start yggdrasild
```
would be the safe bet

Reply:
thanks, updated! :+1: